### PR TITLE
Exposed `get_base` and `get_exp` functions

### PR DESCRIPTION
**Pull request by anutosh491**

### CODE REVIEW

**Comment by anutosh491** [comment]
@isuruf Not sure why some tests fail here. Everything passes for me locally! 
Am I missing some flag while building the project through cmake ?

File: symengine/cwrapper.cpp [hunk 1]
```diff
@@ -578,6 +578,22 @@ CWRAPPER_OUTPUT_TYPE basic_pow(basic s, const basic a, const basic b)
     CWRAPPER_END
 }
 
+CWRAPPER_OUTPUT_TYPE basic_get_base(basic s, const basic a)
```
Comment (isuruf) [inline symengine/cwrapper.cpp:581]:
```suggestion
CWRAPPER_OUTPUT_TYPE basic_pow_get_base(basic s, const basic a)
```

**Review by isuruf — COMMENTED** [review]
This can already be achieved by using `basic_get_args` right?

**Comment by anutosh491** [comment]
> This can already be achieved by using basic_get_args right?

Yes, I missed this. I've made the change.

**Comment by isuruf** [comment]
No, I mean a user could use that method already in the C wrapper and I don't see why we'd need another method to do something similar.

**Comment by anutosh491** [comment]
Okay for some context, Ondrej and I would be adding support for some methods through the Cwrapper for addressing this TODO https://github.com/lcompilers/lpython/issues/2332
We are trying to port some functionalities like the ones we could have through SymPy using Lpython (like `expr.base` and `expr.exp`) so that we could essentially support some modules like series through Lpython.

**Comment by anutosh491** [comment]
Also could you educate me regarding the failing tests ?
When I run ctests locally everything works as required .

**Comment by certik** [comment]
I think we should be designing the Symbolic API to be consistent, and I think we have done a lot of this work in SymEngine already. In particular, I think it's cleaner to access the arguments using `.args`, so I would just use that.

Regarding the failing tests, not sure what is causing it, you would need to debug it locally.

**Comment by certik** [comment]
I think we can close this one. If you think we need it, you can reopen it.

**Comment by anutosh491** [comment]
Sure if we can do without creating these then let's try it out. I just thought of introducing some consistency with sympy but I think we can do without it !

**Comment by certik** [comment]
LPython adds the consistency with SymPy. In SymEngine I think we should strive for a minimal C++ and C API, without duplication.

**Pull request closed without merging**